Use illumos headers as GCC headers fail to behave correctly in 32-bit environment when _KERNEL (and thus _I32LPx) is defined

--- VirtualBox-7.0.18/Config.kmk.orig	2024-05-05 15:05:58.736206131 -0700
+++ VirtualBox-7.0.18/Config.kmk	2024-05-05 15:16:00.225795776 -0700
@@ -5437,7 +5437,7 @@
  TEMPLATE_VBoxR0_TOOL               = $(VBOX_GCC_TOOL)
  TEMPLATE_VBoxR0_CFLAGS             = -fno-pie -nostdinc -g $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_C) \
  	$(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_R0_OPT) $(VBOX_GCC_R0_FP) -fno-strict-aliasing -fno-exceptions \
- 	$(VBOX_GCC_fno-stack-protector) -fno-common -ffreestanding $(VBOX_GCC_fvisibility-hidden) -std=gnu99 $(VBOX_GCC_IPRT_FMT_CHECK)
+ 	$(VBOX_GCC_fno-stack-protector) -fno-common $(VBOX_GCC_fvisibility-hidden) -std=gnu99 $(VBOX_GCC_IPRT_FMT_CHECK)
  TEMPLATE_VBoxR0_CXXFLAGS           = -fno-pie -nostdinc -g $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_CXX) \
  	$(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_R0_OPT) $(VBOX_GCC_R0_FP) -fno-strict-aliasing -fno-exceptions \
  	$(VBOX_GCC_fno-stack-protector) -fno-common $(VBOX_GCC_fvisibility-inlines-hidden) $(VBOX_GCC_fvisibility-hidden) \
@@ -5445,7 +5445,7 @@
  if $(VBOX_GCC_VERSION_CC) >= 40500 # 4.1.2 complains, 4.5.2 is okay, didn't check which version inbetween made it okay with g++.
   TEMPLATE_VBoxR0_CXXFLAGS         += -ffreestanding
  endif
- TEMPLATE_VBoxR0_CFLAGS.amd64       = -m64 -mno-red-zone -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -fasynchronous-unwind-tables -ffreestanding
+ TEMPLATE_VBoxR0_CFLAGS.amd64       = -m64 -mno-red-zone -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -fasynchronous-unwind-tables
  TEMPLATE_VBoxR0_CXXFLAGS.amd64     = -m64 -mno-red-zone -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -fasynchronous-unwind-tables
  TEMPLATE_VBoxR0_CXXFLAGS.freebsd   = -ffreestanding
  if $(VBOX_GCC_VERSION_CC) < 30400
@@ -5844,7 +5844,7 @@
  	-Wmissing-prototypes -Wstrict-prototypes $(VBOX_GCC_fdiagnostics-show-option) \
  	-Wshadow -Wuninitialized -Wunused-function -Wunused-label -Wunused-value -Wunused-variable \
  	-Wformat  -Wimplicit-int -Wimplicit-function-declaration -Werror-implicit-function-declaration -Wparentheses \
- 	-O2 -fno-omit-frame-pointer -ffreestanding -nodefaultlibs -fno-strict-aliasing -fno-common -std=c99
+ 	-O2 -fno-omit-frame-pointer -nodefaultlibs -fno-strict-aliasing -fno-common -std=c99
  TEMPLATE_VBoxR0Drv_CFLAGS.x86         = -m32 -mno-sse -mno-mmx -mno-sse2 -mno-3dnow
  TEMPLATE_VBoxR0Drv_CFLAGS.amd64       = -m64 -mno-sse -mno-mmx -mno-sse2 -mno-3dnow $(VBOX_GCC_msave-args) \
  	-mno-red-zone -mcmodel=kernel $(if-expr defined(VBOX_GCC_USING_SOLARIS_AS),-Wa$(COMMA)-xmodel=kernel,) \
@@ -8660,9 +8660,9 @@
 TEMPLATE_VBoxGuestR3XFree86Mod_SYSSUFF := .o
 TEMPLATE_VBoxGuestR3XFree86Mod_DEFS     = $(TEMPLATE_VBoxGuestR3ExeBase_DEFS) VBOX_GUESTR3XF86MOD RTMEM_NO_WRAP_TO_EF_APIS
 TEMPLATE_VBoxGuestR3XFree86Mod_CFLAGS   = $(filter-out $(VBOX_GCC_SANITIZER_FLAGS),$(TEMPLATE_VBoxGuestR3ExeBase_CFLAGS)) \
-	-fno-pie -fno-merge-constants -std=c99 -ffreestanding
+	-fno-pie -fno-merge-constants -std=c99
 TEMPLATE_VBoxGuestR3XFree86Mod_CXXFLAGS = $(filter-out $(VBOX_GCC_SANITIZER_FLAGS),$(TEMPLATE_VBoxGuestR3ExeBase_CXXFLAGS)) \
-	-fno-pie -fno-merge-constants -ffreestanding
+	-fno-pie -fno-merge-constants
 TEMPLATE_VBoxGuestR3XFree86Mod_CFLAGS.linux             = $(VBOX_GCC_fno-stack-protector) # Remove symvers.h
 TEMPLATE_VBoxGuestR3XFree86Mod_CXXFLAGS.linux           = $(VBOX_GCC_fno-stack-protector) # Remove symvers.h
 TEMPLATE_VBoxGuestR3XFree86Mod_LDFLAGS.$(KBUILD_TARGET) = $(NO_SUCH_VARIABLE)
